api/src/app.ts

```diff
@@ -210,9 +212,16 @@ export const build = async (
   if (FCC_ENABLE_DEV_LOGIN_MODE) {
     void fastify.register(devLoginCallback, { prefix: '/auth' });
   }
+
+  // Stripe plugin
+  const stripe = new Stripe(STRIPE_SECRET_KEY, {
+    apiVersion: '2020-08-27',
+    typescript: true
+  });
```

Comment:
This is looking great, Ahmad! The only thing that stood out was the creation of the stripe instance here. Since you've already mocked it in the test, is there any reason to inject it into dontateRoutes?

Reply:
I ran to problems customizing the the function returns of the mocked Stripe instance. That is why I decided to pass the instance instead of creating it in the route.

Reply:
Okay. It took a bit of messing around, but I figured out how to get it to work when creating it in the donate plugin.

Since we were only using dependency injection to inject mocks, it seems better (more modular, better separation of concerns) to keep the stripe instance in the plugin.

If you agree, do you mind me pushing a commit?

Reply:
Of course. Curious to see what you did.